process_data.go

```diff
@@ -198,15 +198,19 @@ func (dsp *DataStreamProcessor) AnalyzeData(records []*DataRecord) {
 		var val float64        // used to calculate pretrigger mean, then reused in the next loop
 		var valPTDelta float64 // used to calculate pretrigger delta
 		// slope = dot(x,y.-y[0])/z where z = dot(x,x) and x = [0, 1, 2, ..., N-1]/(N-1), where .-y[0] is elementwise subtraction of the first element
-		np := rec.presamples
-		a := float64(np*(np-1)*(2*np-1)) / float64(6) / float64((np - 1)) // a is the sum of np^2 from 0 to np-1 over (np-1)
-		// a converts valPTDelta to the change over the length of the pretrigger
-		for i := 0; i < rec.presamples; i++ {
+		npre := rec.presamples
+		d0 := dataVec.AtVec(0)
+		xmean := float64(npre-1)*0.5
+		for i := 0; i < npre; i++ {
 			val += dataVec.AtVec(i)
-			valPTDelta += (dataVec.AtVec(i) - dataVec.AtVec(0)) * float64(i)
+			valPTDelta += (dataVec.AtVec(i) - d0) * (float64(i) - xmean)
 		}
-		ptm := val / float64(rec.presamples)
-		rec.pretrigDelta = valPTDelta / a
+		ptm := val / float64(npre)
+		rec.pretrigDelta = valPTDelta * 12.0 / float64(npre*(npre+1))
+		if npre <= 1 {
```

Comment:
The other assignment should be an else clause here

Reply:
Okay. See next commit